pkg/odo/cli/project/create.go

```diff
@@ -0,0 +1,87 @@
+package project
+
+import (
+	"fmt"
+
+	"github.com/redhat-developer/odo/pkg/log"
+	"github.com/redhat-developer/odo/pkg/odo/genericclioptions"
+	"github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/redhat-developer/odo/pkg/project"
+	"github.com/spf13/cobra"
+
+	ktemplates "k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+)
+
+const createRecommendedCommandName = "create"
+
+var (
+	createExample = ktemplates.Examples(`
+	# Create a new project
+	%[1]s myproject
+	`)
+
+	createLongDesc = ktemplates.LongDesc(`Create a new project`)
+
+	createShortDesc = `Create a new project`
+)
+
+// ProjectCreateOptions encapsulates the options for the odo project create command
+type ProjectCreateOptions struct {
+
+	// name of the project
+	projectName string
+
+	// generic context options common to all commands
+	*genericclioptions.Context
+}
+
+// NewProjectCreateOptions creates a ProjectCreateOptions instance
+func NewProjectCreateOptions() *ProjectCreateOptions {
+	return &ProjectCreateOptions{}
+}
+
+// Complete completes ProjectCreateOptions after they've been created
+func (pco *ProjectCreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	pco.projectName = args[0]
+	pco.Context = genericclioptions.NewContext(cmd)
+	return
+}
+
+// Validate validates the parameters of the ProjectCreateOptions
+func (pco *ProjectCreateOptions) Validate() (err error) {
```

Comment:
`o`

Reply:
There is a reason for this. Using `pco` would help refactoring in case of the struct name change in the future.

Reply:
The name of the variable would only help if you're doing text replacement and not using a specific tool (IDE)…